docs/semgrep-ci/network-broker.md

```diff
@@ -0,0 +1,134 @@
+      peers:
+        - publicKey: 4EqJwDZ8X/qXB5u3Wpo2cxnKlysec93uhRvGWPix0lg=
+          endpoint: wireguard.semgrep.dev:51820
+          allowedIps: fdf0:59dc:33cf:9be9:0000:0000:0000:0001/128
+    heartbeat:
+      url: http://[fdf0:59dc:33cf:9be9:0000:0000:0000:0001]/ping
+    allowlist: []
+    gitlab:
+      baseUrl: <https://gitlab.xxxx.net/api/v4>
+      token: YOUR_TOKEN
+  ```
+
+  The `publicKey` should be entered precisely as follows:
+
+  ```console
+  4EqJwDZ8X/qXB5u3Wpo2cxnKlysec93uhRvGWPix0lg=
+  ```
+
+1. The broker requires a WireGuard keypair to establish a secure connection. To generate the public key:
+
+   1. Get the [network broker version](https://github.com/semgrep/semgrep-network-broker/pkgs/container/semgrep-network-broker) you want to use. The format should be similar to `v0.14.0`.
+
+   1. Run the following in the CLI to generate the public key, replacing the placeholder with the network broker version number:
+  <pre class="language-console"><code>docker run ghcr.io/semgrep/semgrep-network-broker:<span className="placeholder">VERSION_NUMBER</span> genkey</code></pre>
+
+1. Run the following in the CLI to generate the private key, replacing the placeholders with the public key you generated in the previous step and the network broker version number:
+
+  <pre class="language-console"><code>echo `<span className="placeholder">PUBLIC_KEY</span>` | sudo docker run -i ghcr.io/semgrep/semgrep-network-broker:<span className="placeholder">VERSION_NUMBER</span> pubkey</code></pre>
+
+  :::info Key sharing 
+  Your public key is safe to share. Do **not** share your private key with anyone, including Semgrep.
+  :::
+
+1. Update the `config.yaml` file with the private key:
+
+  ```yaml
+  inbound:
+    wireguard:
+      localAddress: fdf0:59dc:33cf:9be8:yyyy:0:1
+      privateKey: YOUR_PRIVATE_KEY
+      ...
+  ```
+
+1. Add the generated public key to the Semgrep Cloud Platform:
+
+   1. Log into Semgrep Cloud Platform.
+   2. Navigate to **Settings** > **Broker**.
+   3. Paste your public key and click **Add Public Key**.
+
+   ![Screenshot of Semgrep Cloud Platform's Network Broker page](/img/scp-broker.png#md-width)
+
+2. Update the `config.yaml` with your GitLab or GitHub URL:
+
+  ```yaml
+  # for GitLab 
+  gitlab:
+    baseUrl: <https://gitlab.exampleCo.net/api/v4>
+    token: YOUR_TOKEN
+
+  # for GitHub 
+  gitlab:
+    baseUrl: <https://github.exampleCo.com/api/v3>
+    token: YOUR_TOKEN
+  ```
+
+1. [Create a GitLab personal access token (PAT)](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token) with the `api` scope, and add the token to `config.yaml` using the `token` field:
+
+  ```yaml
+  gitlab:
+    baseUrl: <https://gitlab.xxxx.net/api/v4>
+    token: YOUR_TOKEN
+  ```
+
+1. [Convert your deployment ID to hexadecimal](https://www.rapidtables.com/convert/number/decimal-to-hex.html) for use in creating your deployment's local address:
```

Comment:
I'm always a little nervous about linking to third-party tools in our docs in case something happens to them. Maybe we could phrase this more carefully, like 
```suggestion
1. Convert your deployment ID to hexadecimal for use in creating your deployment's local address. You can use a tool such as [Decimal to Hexadecimal converter]((https://www.rapidtables.com/convert/number/decimal-to-hex.html)) if needed.
```